docs/source/metrics.rst

```diff
@@ -0,0 +1,141 @@
+=======
+Metrics
+=======
+BentoML now supports all metrics found in Prometheus Client. 
+User can create custom metrics and attach them to API endpoints and access them through the ``/metrics`` endpoint.
```

Comment:
I would start with 

Preface -> Why you do you need this -> a e2e example -> dive into details.


Also ``/metrics`` only true for HTTP. for gRPC it should be `0.0.0.0:3001`